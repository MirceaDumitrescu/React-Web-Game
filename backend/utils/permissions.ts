export const permissions = {
    user: 'user',
    admin: 'admin'
};
