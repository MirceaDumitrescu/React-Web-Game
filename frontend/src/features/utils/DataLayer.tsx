import { configureStore } from '@reduxjs/toolkit'
import { Provider } from 'react-redux'
import userReducer from '../../features/reducers/loginStatusReducer'
import { FC } from 'react'

const store = configureStore({
  reducer: {
    user: userReducer,
  },
})

type Props = {
  children: React.ReactNode
}

const DataLayerComponent: FC<Props> = ({ children }) => {
  return <Provider store={store}>{children}</Provider>
}

export default DataLayerComponent
