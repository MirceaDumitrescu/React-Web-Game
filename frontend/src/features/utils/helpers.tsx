import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import DataLayerComponent from './DataLayer'

const createStoreComponent = (Component: React.ElementType) => {
  // eslint-disable-next-line react/display-name
  return () => {
    return (
      <MemoryRouter>
        <DataLayerComponent>
          <Component />
        </DataLayerComponent>
      </MemoryRouter>
    )
  }
}

export default createStoreComponent
