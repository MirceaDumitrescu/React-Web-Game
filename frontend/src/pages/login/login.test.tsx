import LoginComponent from './login'
import { screen } from '@testing-library/react'
import { renderWithProviders } from '../../features/utils/test-utils'
import '@testing-library/jest-dom'
import { MemoryRouter } from 'react-router-dom'

describe('Login Component', () => {
  beforeEach(() => {
    renderWithProviders(
      <MemoryRouter>
        <LoginComponent />
      </MemoryRouter>,
    )
  })

  it('should render the login component', () => {
    expect(screen.getByTestId('login-form')).toBeInTheDocument()
  })

  it('should have an email field and a password field', () => {
    expect(screen.getByTestId('email')).toBeInTheDocument()
    expect(screen.getByTestId('password')).toBeInTheDocument()
  })

  it('should have a login button', () => {
    expect(screen.getByTestId('form-button')).toBeInTheDocument()
  })

  it('should have a link to register', () => {
    expect(screen.getByText('Missing an account? Register here!')).toBeInTheDocument()
  })
})
