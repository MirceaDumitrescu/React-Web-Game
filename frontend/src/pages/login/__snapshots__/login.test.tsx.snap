// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="form__login"
    data-testid="login-form"
  >
    <h4
      class="form__title"
    >
      Login Form
    </h4>
    <div
      class="form__group"
    >
      <form
        class="form"
      >
        <div
          class="form__group"
        >
          <label>
            Email adress
          </label>
          <input
            class="form__group__input"
            data-testid="email"
            name="email"
            type="email"
            value=""
          />
        </div>
        <div
          class="form__group"
        >
          <label>
            Password
          </label>
          <input
            class="form__group__input"
            data-testid="password"
            name="password"
            type="password"
            value=""
          />
        </div>
        <div
          class="form__group"
        >
          <label />
          <input
            class="form__btn"
            data-testid="button"
            name="button"
            type="submit"
            value="Login"
          />
        </div>
      </form>
      <div
        class="form__group"
      >
        <button
          class="form__redirect"
          data-testid="form-button"
        >
          <a
            class="form__redirect__link"
            href="/register"
          >
              Missing an account? Register here!
          </a>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="form__login"
    data-testid="login-form"
  >
    <h4
      class="form__title"
    >
      Login Form
    </h4>
    <div
      class="form__group"
    >
      <form
        class="form"
      >
        <div
          class="form__group"
        >
          <label>
            Email adress
          </label>
          <input
            class="form__group__input"
            data-testid="email"
            name="email"
            type="email"
            value=""
          />
        </div>
        <div
          class="form__group"
        >
          <label>
            Password
          </label>
          <input
            class="form__group__input"
            data-testid="password"
            name="password"
            type="password"
            value=""
          />
        </div>
        <div
          class="form__group"
        >
          <label />
          <input
            class="form__btn"
            data-testid="button"
            name="button"
            type="submit"
            value="Login"
          />
        </div>
      </form>
      <div
        class="form__group"
      >
        <button
          class="form__redirect"
          data-testid="form-button"
        >
          <a
            class="form__redirect__link"
            href="/register"
          >
              Missing an account? Register here!
          </a>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
